File: TaskManager/TaskManager/Command/UpdateIssueCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TaskManager
{
    public class UpdateIssueCommand : ICommand
    {
        private IssueRepository _repo;
        private Issue _oldIssue;
        private Issue _newIssue;

        public UpdateIssueCommand(IssueRepository repo, Issue oldIssue, Issue newIssue)
        {
            _repo = repo;
            _oldIssue = oldIssue;
            _newIssue = newIssue;
        }

        public void Execute()
        {
            _repo.UpdateIssue(_newIssue);
        }

        public void Undo()
        {
            _repo.UpdateIssue(_oldIssue);
        }
    }
}

File: TaskManagerSolution/src/main/java/com/taskmanager/command/UpdateIssueCommand.java
package com.taskmanager.command;

import com.taskmanager.model.Issue;
import com.taskmanager.service.IssueService;

public class UpdateIssueCommand implements Command {
    private IssueService issueService;
    private Issue issue;

    public UpdateIssueCommand(IssueService issueService, Issue issue) {
        this.issueService = issueService;
        this.issue = issue;
    }

    @Override
    public void execute() {
        issueService.updateIssue(issue);
    }

    @Override
    public void undo() {

    }
}

































